rest/config.go

```diff
@@ -1023,10 +1023,15 @@ func RunServer(config *ServerConfig) {
 	}
 
 	sc := NewServerContext(config)
-	for _, dbConfig := range config.Databases {
-		if _, err := sc.AddDatabaseFromConfig(dbConfig); err != nil {
-			base.Fatalf("Error opening database %s: %+v", base.MD(dbConfig.Name), err)
-		}
+	dbContextMap, err, dbName := addDatabaseFromConfig(sc, config.Databases)
```

Comment:
As discussed, let's switch to the following approach:
- the only change to RunServer is to call a new 'validateServerContext(serverContext)' after the iteration to add databases 
- validateServerContext logs to warn directly
- validateServerContext will return []error, where errors are typed errors that can be used by test code 
- RunServer ignores the returned value

Reply:
This change is implemented in latest revision.